src/main/java/com/hazelcast/gcp/GcpDiscoveryStrategyFactory.java

```diff
@@ -50,4 +50,47 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
         }
         return result;
     }
+
+    @Override
```

Comment:
Could you add a javodoc here?

It should include:
1. Our reasoning about using `/etc/resolv.conf` and `/etc/hosts` (it's not an official way, but good enough as a heuristic to check auto-detection)
2. Link to the official doc: https://cloud.google.com/compute/docs/instances/managing-instances#dmi

Reply:
I added it